ssl/statem/statem_lib.c

```diff
@@ -704,6 +704,19 @@ static int version_cmp(const SSL *s, int a, int b)
     return DTLS_VERSION_LT(a, b) ? -1 : 1;
 }
 
+int ssl_strict_version_check(const SSL *s, int version)
```

Comment:
We don't want to offer a session for a version we are not prepared to accept from the server. The ssl_method_error function also does the check for SSL_OP_NO_*.